client/allocrunner/taskrunner/task_runner_test.go

```diff
@@ -2866,6 +2866,75 @@ func TestTaskRunner_BaseLabels(t *testing.T) {
 	require.Equal(alloc.Namespace, labels["namespace"])
 }
 
+// TestTaskRunner_BaseLabels_IncludesMetadata tests that the base labels include
+// the metadata fields
+func TestTaskRunner_BaseLabels_IncludesMetadata(t *testing.T) {
+	ci.Parallel(t)
+	require := require.New(t)
+
+	alloc := mock.BatchAlloc()
+	alloc.Namespace = "not-default"
+	job := alloc.Job
+	job.Meta = map[string]string{"owner": "HashiCorp", "my-key": "my-value", "some_dynamic_value": "now()"}
+	task := job.TaskGroups[0].Tasks[0]
+	task.Driver = "raw_exec"
+	task.Config = map[string]interface{}{
+		"command": "whoami",
+	}
+
+	config, cleanup := testTaskRunnerConfig(t, alloc, task.Name, nil)
+	defer cleanup()
+
+	config.ClientConfig.IncludeMetadataInMetrics = true
+
+	tr, err := NewTaskRunner(config)
+	require.NoError(err)
+
+	labels := map[string]string{}
+	for _, e := range tr.baseLabels {
+		labels[e.Name] = e.Value
+	}
+
+	require.Equal(task.Meta["owner"], labels["owner"])
+	require.Equal(task.Meta["some_dynamic_value"], labels["some_dynamic_value"])
+	require.Equal(task.Meta["my-key"], labels["my_key"])
```

Comment:
More `shoenig/test` migration. But also, these test assertions could be constants so that you know you're not introducing test dependency on the internals:

```suggestion
	must.Eq(t, "HashiCorp", labels["owner"])
	must.Eq(t, "now()", labels["some_dynamic_value"])
	must.Eq(t, "my-value", labels["my_key"])
```

Alternately, you could skip iterating the `baseLabels` and just `must.Eq` the whole map.

Reply:
I have opted for `must.Eq` as I didn't want to test the base labels like `alloc_id` and others as part of this